plugins/modules/network_compliance_workflow_manager.py

```diff
@@ -1060,30 +976,49 @@ def run_compliance(self, run_compliance_params):
             This method initiates a compliance check operation in Cisco DNA Center by calling the "run_compliance" function
             from the "compliance" family of APIs. It passes the provided parameters and updates the result accordingly.
         """
-
         # Execute the compliance check operation
-        try:
-            response = self.dnac_apply["exec"](
-                family="compliance",
-                function="run_compliance",
-                params=run_compliance_params,
-                op_modifies=True,
-            )
-            self.log("Response received post 'run_compliancee' API call is {0}".format(str(response)), "DEBUG")
+        device_uuids = run_compliance_params.get("deviceUuids")
```

Comment:
That is already being handled here:
        if not mgmt_ip_to_instance_id_map:
            # Log an error message if mgmt_ip_to_instance_id_map is empty
            msg = ("No device UUIDs were fetched for network compliance operations with the provided IP address(es): {0} "
                   "or site name: {1}. This could be due to Unreachable devices or access points (APs).").format(", ".join(ip_address_list), site_name)
            self.log(msg, "ERROR")
            self.module.fail_json(msg)


Before adding them to the run_compliance_params

Reply:
This API could be called anywhere, hence it is better to check this API as well.

Reply:
Doing the validation before, so the module doesn't proceed only if there are no device_uuids provided for module execution.